src/engine/controls/loopingcontrol.cpp

```diff
@@ -527,24 +527,30 @@ double LoopingControl::getSyncPositionInsideLoop(double dRequestedPlaypos, doubl
     return dSyncedPlayPos;
 }
 
+void LoopingControl::setBeatLoop(double startPosition, bool enabled) {
+    VERIFY_OR_DEBUG_ASSERT(startPosition != Cue::kNoPosition) {
+        return;
+    }
+
+    mixxx::BeatsPointer pBeats = m_pBeats;
+    if (!pBeats) {
+        return;
+    }
+
+    double beatloopSize = m_pCOBeatLoopSize->get();
+    double endPosition = pBeats->findNBeatsFromSample(startPosition, beatloopSize);
```

Comment:
Not realtime safe... but that's a [job for another PR](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/getting.20locks.20out.20of.20Beats).

Reply:
Done.